src/index.ts

```diff
@@ -367,64 +416,27 @@ export abstract class DetSysAction {
     }
   }
 
-  addFact(key: string, value: string | boolean): void {
-    this.facts[key] = value;
-  }
-
-  getDiagnosticsUrl(): URL | undefined {
-    return this.actionOptions.diagnosticsUrl;
-  }
-
-  getUniqueId(): string {
-    return (
-      this.identity.run_differentiator ||
-      process.env.RUNNER_TRACKING_ID ||
-      randomUUID()
-    );
-  }
-
-  getCorrelationHashes(): correlation.AnonymizedCorrelationHashes {
-    return this.identity;
-  }
-
-  recordEvent(eventName: string, context: Record<string, unknown> = {}): void {
-    this.events.push({
-      event_name: `${this.actionOptions.eventPrefix}${eventName}`,
-      context,
-      correlation: this.identity,
-      facts: this.facts,
-      timestamp: new Date(),
-      uuid: randomUUID(),
-    });
-  }
-
   /**
-   * Fetches a file in `.xz` format, imports its contents into the Nix store,
-   * and returns the path of the executable at `/nix/store/STORE_PATH/bin/${bin}`.
-   */
-  async unpackClosure(bin: string): Promise<string> {
-    const artifact = this.fetchArtifact();
-    const { stdout } = await promisify(exec)(
-      `cat "${artifact}" | xz -d | nix-store --import`,
-    );
-    const paths = stdout.split(os.EOL);
-    const lastPath = paths.at(-2);
-    return `${lastPath}/bin/${bin}`;
-  }
-
-  /**
-   * Fetch an artifact, such as a tarball, from the URL determined by the `source-*`
-   * inputs and other factors.
+   * Fetch an artifact, such as a tarball, from the URL determined by the
+   * `source-*` inputs.
```

Comment:
nit:
```suggestion
   * Fetch an artifact, such as a tarball, from the location determined by the
   * `source-*` inputs.
```

(since none of them aside from `source-url` are actually URLs, and `source-binary` doesn't point at a URL unlike e.g. `source-pr` pointing to `i.d.s/pr/#`)

Reply:
Will do a quick follow-up